File: RL_Maggie/Code/01_copy_raw_files.R
## 01_copy_raw_files.R ---------------------------------------------------
## Mirror RL csvs from the shared drive to LOCAL_DATA_DIR

copy_files_from_rl <- function(source_dir, dest_dir) {
  dir_create(dest_dir, recurse = TRUE)
  subdirs      <- dir_ls(source_dir, recurse = TRUE, type = "directory")
  rl_subdirs   <- subdirs[str_detect(subdirs, "/sub_\\d+/RL$")]
  
  for (subdir in rl_subdirs) {
    files <- dir_ls(subdir, type = "file")
    matching <- files[str_detect(path_file(files), "^sub_\\d+_Reversal_.*\\.csv$")]
    file_copy(matching, path(dest_dir, path_file(matching)), overwrite = TRUE)
  }
}

copy_files_from_rl(RAW_SHARED_DIR, LOCAL_DATA_DIR)

File: RL_Maggie/python/01_copy_raw_files.py
from pathlib import Path
import shutil
from importlib import import_module

setup = import_module('RL_Maggie.python.00_setup')
RAW_SHARED_DIR = setup.RAW_SHARED_DIR
LOCAL_DATA_DIR = setup.LOCAL_DATA_DIR


def copy_files_from_rl(source_dir: Path, dest_dir: Path) -> None:
    dest_dir.mkdir(parents=True, exist_ok=True)
    subdirs = list(source_dir.rglob('*'))
    rl_subdirs = [d for d in subdirs if d.is_dir() and d.name == 'RL']

    for subdir in rl_subdirs:
        for file in subdir.iterdir():
            if file.is_file() and file.name.startswith('sub_') and file.name.endswith('.csv'):
                shutil.copy2(file, dest_dir / file.name)


if __name__ == '__main__':
    copy_files_from_rl(RAW_SHARED_DIR, LOCAL_DATA_DIR)
